src/main/java/model/Player.java

```diff
@@ -302,17 +301,20 @@ public void checkCollideMonster(final Monster monster) {
      * This method is used when the character is killed.
      */
     public void die() {
-        if (this.getLives() <= 1 && !this.isDead) {
-            this.isDead = true;
-            spriteBase.setDx(0);
-            spriteBase.setDy(0);
+        this.loseLife();
+        this.scorePoints(Settings.POINTS_PLAYER_DIE);
+        spriteBase.setImage("/Bub" + playerNumber + "Death.png");
+        spriteBase.setDx(0);
+        spriteBase.setDy(0);
+
+        if (this.getLives() == 0) {
             counter = 0;
-            spriteBase.setImage("/Bub" + playerNumber + "Death.png");
+
```

Comment:
Maybe add here:
```
spriteBase.setX(xStartLocation);
spriteBase.setY(yStartLocation);
```
to set a died player at his starting position. To make this work nice you have to adjust the starting position of map2.txt, and map3.txt to place a died player on a platform (map2) and not make him be halfway in the wall (map3)